fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -430,55 +432,74 @@ private CompletableFuture<Void> setIndexingTypeOrThrow(FDBRecordStore store, boo
     }
 
     @Nonnull
-    private CompletableFuture<Void> setIndexingTypeOrThrow(FDBRecordStore store, boolean continuedBuild, Index index, IndexBuildProto.IndexBuildIndexingStamp indexingTypeStamp) {
+    private CompletableFuture<Void> setIndexingTypeOrThrow(FDBRecordStore store, boolean continuedBuild, Index index, IndexBuildProto.IndexBuildIndexingStamp newStamp) {
         if (forceStampOverwrite && !continuedBuild) {
             // Fresh session + overwrite = no questions asked
-            store.saveIndexingTypeStamp(index, indexingTypeStamp);
+            store.saveIndexingTypeStamp(index, newStamp);
             return AsyncUtil.DONE;
         }
         return store.loadIndexingTypeStampAsync(index)
                 .thenCompose(savedStamp -> {
                     if (savedStamp == null) {
-                        if (continuedBuild && indexingTypeStamp.getMethod() !=
+                        if (continuedBuild && newStamp.getMethod() !=
                                               IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
                             // backward compatibility - maybe continuing an old BY_RECORD session
                             return isWriteOnlyButNoRecordScanned(store, index)
-                                    .thenCompose(noRecordScanned -> throwAsByRecordsUnlessNoRecordWasScanned(noRecordScanned, store, index, indexingTypeStamp));
+                                    .thenCompose(noRecordScanned -> throwAsByRecordsUnlessNoRecordWasScanned(noRecordScanned, store, index, newStamp));
                         }
                         // Here: either not a continuedBuild (new session), or a BY_RECORD session (allowed to overwrite the null stamp)
-                        store.saveIndexingTypeStamp(index, indexingTypeStamp);
+                        store.saveIndexingTypeStamp(index, newStamp);
                         return AsyncUtil.DONE;
                     }
                     // Here: has non-null type stamp
-                    if (indexingTypeStamp.equals(savedStamp)) {
+                    if (newStamp.equals(savedStamp)) {
                         // A matching stamp is already there - One less thing to worry about
                         return AsyncUtil.DONE;
                     }
-                    if (continuedBuild && shouldAllowTakeoverContinue(indexingTypeStamp, savedStamp)) {
-                        // Special case: partly built with multi target, but may be continued indexing on its own
-                        store.saveIndexingTypeStamp(index, indexingTypeStamp);
+                    if (isTypeStampBlocked(savedStamp) && !policy.shouldAllowUnblock(savedStamp.getBlockID())) {
+                        // Indexing is blocked
+                        throw newPartlyBuiltException(continuedBuild, savedStamp, newStamp, index);
+                    }
+                    if (areSimilar(newStamp, savedStamp)) {
```

Comment:
Why are these two no longer checking `continueBuild`?

Reply:
For these two, it doesn't matter. If blocked or similar it can either continue or redo from scratch.